tests/test_algo.py

```diff
@@ -166,3 +167,27 @@ def test_execute_predict(workdir, create_models):
     with open(pred_path, 'r') as f:
         pred = json.load(f)
     assert pred == 'XXX'
+
+
+def test_models_generator(mocker, workdir, create_models):
```

Comment:
Could be factorized using two parameters: `use_generator`, `model_argument_type` ?

Reply:
I could but I would lose the final `assert models_1 == list(models_2)`

Reply:
I think this check is not required, the goal of this test is to check the type of the models argument.

If we really want to do this check we could check directly the value of `models`. It looks `create_models` method returns as first  arg the values inside the models.

Reply:
Ok, I parametrized the test, can you re-review?